File: solutions/HJK9810/2026-09-07/low/1959.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

class Solution {
    private int compare(int[] maxList, int[] minList) {
        int loopSize = maxList.length - minList.length + 1;
        int minSize = minList.length;
        int maxTotal = Integer.MIN_VALUE;

        for (int start = 0; start < loopSize; start++) {
            int total = 0;
            for (int idx = 0; idx < minSize; idx++) {
                total += maxList[idx + start] * minList[idx];
            }
            maxTotal = Math.max(maxTotal, total);
        }

        return maxTotal;
    }

    public static void main(String args[]) throws Exception {
        Solution sol = new Solution();

        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(input.readLine());
        StringTokenizer st;

        for(int test_case = 1; test_case <= T; test_case++) {
            st = new StringTokenizer(input.readLine());
            int N = Integer.parseInt(st.nextToken());
            int M = Integer.parseInt(st.nextToken());
            int[] A = new int[N];
            int[] B = new int[M];

            st = new StringTokenizer(input.readLine());
            for (int idx = 0; idx < N; idx++) {
                A[idx] = Integer.parseInt(st.nextToken());
            }
            st = new StringTokenizer(input.readLine());
            for (int idx = 0; idx < M; idx++) {
                B[idx] = Integer.parseInt(st.nextToken());
            }

            int result = N < M ? sol.compare(B, A) : sol.compare(A, B);
            System.out.printf("#%d %d\n", test_case, result);
        }
    }
}

File: solutions/HJK9810/2026-09-07/low/1959.py
def algorithm():
    N, M = map(int, input().split())
    A = list(map(int, input().split()))
    B = list(map(int, input().split()))

    loop_size = abs(M - N) + 1
    min_size = min(N, M)
    max_total = float('inf') * -1

    for start in range(loop_size):
        total = 0
        for idx in range(min_size):
            if N < M:
                total += (A[idx] * B[idx + start])
            else:
                total += (A[idx + start] * B[idx])
        max_total = max(max_total, total)

    return max_total

T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    print(f"#{test_case} {algorithm()}")
